sysfs/src/lib.rs

```diff
@@ -1,5 +1,7 @@
 pub use sysfs_lib as lib;
 pub use sysfs_lib::{Error, Result};
 
+extern crate self as sysfs;
```

Comment:
What does this accomplish?

Reply:
it makes `sysfs` available inside `sysfs`.

Problem is, I'm not sure I fully understood your question, so I just tried fixing the compilation errors.

Reply:
Well this branch was my first attempt, or at least, I think it was but I got it out of a random stash. The question was really open ended which is why you and togglebit had trouble with me. I kinda just wanted to make sure that I set up the workspace correctly. I wanted two nested workspaces, but having a package-workspace inside of a workspace is not allowed. So I guess I'll just keep what I have on master with the two top level crates completely separated.